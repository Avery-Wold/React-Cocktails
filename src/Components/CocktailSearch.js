import React, { Component } from 'react';
import '../App.scss';
import { updateDrinkName } from '../Actions/UpdateInfo';
import { getDrink } from '../Actions/DrinkInfo';
import { connect } from 'react-redux';
import CocktailInput from './CocktailInput';
import FormErrors from './FormErrors';

export class CocktailSearch extends Component {
    constructor(props) {
        super(props); 
        this.state = {
            formErrors: {
              validCocktail: ''
            },
            isCocktailValid: false,
            showFormErrors: false
        }       
    }

    getCocktail = (e) => {
        e.preventDefault();
        let newFormErrors = this.state.formErrors;
        let validCocktail = this.state.isCocktailValid;    
        const {dispatch} = this.props;
        if (this.props.drink !== ""){
            dispatch((getDrink(this.props.drink))); 
        }
        else {
            newFormErrors.validCocktail = 'Please enter a valid drink name';
            this.setState({formErrors: newFormErrors, showFormErrors: true, isCocktailValid: validCocktail});
        }
    }
    
    handleChange = (event) => {
        event.preventDefault();
        this.setState({showFormErrors: false});
        var cocktail = event.target.value;
        var searchString = cocktail.replace(/\s/g, "+");
        console.log(searchString);
        const { dispatch } = this.props;
        dispatch(updateDrinkName(cocktail));
    }

    render() {
        return (
            <div className="drink-selection">
                <p>
                Already know what you want to make?
                </p>
                <p>
                <CocktailInput value={this.props.drink} name="drink" onChange={this.handleChange} />
                <button 
                    className="btn"
                    onClick={e => this.getCocktail(e)}>
                    Get Drink
                </button>
                </p>
                {(!this.state.isCocktailValid) && this.state.showFormErrors && <FormErrors errors={this.state.formErrors} />}
            </div>
        )
    }
}
  
const mapStateToProps = (state) => {
    return {
        drink: state.DrinkInfo.drink
    }
}

export default connect(mapStateToProps) (CocktailSearch)
