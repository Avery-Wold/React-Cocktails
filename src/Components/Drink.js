import React, { Component } from 'react';
import '../App.scss';

export class Drink extends Component {
    render() {
        return (
            <div>
                {this.props.drinks.map((drinks) => {
                    return(
                        <div className="drink">
                            <img src={drinks.strDrinkThumb} className="drink-img" alt="cocktail"/>
                            <div className="table">
                                <h1>
                                    {drinks.strDrink}
                                </h1>
                                <h2>
                                    What you'll need:
                                </h2>
                                <p>
                                    {drinks.strGlass}
                                </p>
                                <p>
                                    {drinks.strMeasure1} {drinks.strIngredient1}
                                </p>
                                <p>
                                    {drinks.strMeasure2} {drinks.strIngredient2}
                                </p>
                                <p>
                                    {drinks.strMeasure3} {drinks.strIngredient3}
                                </p>
                                <p>
                                    {drinks.strMeasure4} {drinks.strIngredient4}
                                </p>
                                <p>
                                    {drinks.strMeasure5} {drinks.strIngredient5}
                                </p>
                                <p>
                                    {drinks.strMeasure6} {drinks.strIngredient6}
                                </p>
                                <p>
                                    {drinks.strMeasure7} {drinks.strIngredient7}
                                </p>
                                <p>
                                    {drinks.strMeasure8} {drinks.strIngredient8}
                                </p>
                                <p>
                                    {drinks.strMeasure9} {drinks.strIngredient9}
                                </p>
                                <p>
                                    {drinks.strMeasure10} {drinks.strIngredient10}
                                </p>
                                <p>
                                    {drinks.strMeasure11} {drinks.strIngredient11}
                                </p>
                                <p>
                                    {drinks.strMeasure12} {drinks.strIngredient12}
                                </p>
                                <p>
                                    {drinks.strMeasure13} {drinks.strIngredient13}
                                </p>
                                <p>
                                    {drinks.strMeasure14} {drinks.strIngredient14}
                                </p>
                                <p>
                                    {drinks.strMeasure15} {drinks.strIngredient15}
                                </p>
                                <h2>
                                    Instructions:
                                </h2>
                                <p>
                                    {drinks.strInstructions}
                                </p>
                            </div>
                        </div>
                    );
                })}
            </div>
        )
    }
}