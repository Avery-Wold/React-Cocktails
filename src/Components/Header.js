import React, { Component } from 'react';
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem, NavLink } from 'reactstrap';
import 'bootstrap/dist/css/bootstrap.css';
import '../App.scss';

export default class Example extends Component {
  constructor(props) {
    super(props);

    this.state = {
      collapsed: true
    };
  }

  toggleNavbar = () => {
    this.setState(state => ({
      collapsed: !this.state.collapsed
    }));
  }

  render() {
    return (
      <div className="navbar navbar-expand-md bg-dark navbar-dark">
        <Navbar >
          <NavbarBrand href="/" className="title">Making Cocktails With Avery</NavbarBrand>
          <div className="pull-right">
          <NavbarToggler onClick={this.toggleNavbar} className="navbar-collapse"/>
          <Collapse isOpen={!this.state.collapsed} >
            <Nav navbar>
              <NavItem>
                <NavLink href="https://github.com/Avery-Wold/React-Masters-Spring-2019">GitHub</NavLink>
              </NavItem>
            </Nav>
          </Collapse>
          </div>
        </Navbar>
      </div>
    );
  }
}
