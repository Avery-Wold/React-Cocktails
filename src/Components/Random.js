import React, { Component } from 'react';
import '../App.scss';
import { Drink } from './Drink';
import { getRandom } from '../Actions/DrinkInfo';
import { connect } from 'react-redux';

class Random extends Component {
    constructor(props) {
        super(props);
        
        this.GetRandomCocktail = this.GetRandomCocktail.bind(this);
    }

    GetRandomCocktail(e) {
        e.preventDefault();
        const {dispatch} = this.props;
        dispatch((getRandom()));    
    }

    render() {
        return (
            <div className="drink-selection">
                <p>
                Just want a random drink?
                </p>
                <p>
                <button className="btn" onClick={e => this.GetRandomCocktail(e)}>
                    Get Random Drink
                </button>
                </p>
            </div>
        )
    }
}
  
const mapStateToProps = (state) => {
    return {
        drink: state.DrinkInfo.drink,
        glass: state.DrinkInfo.glass,
        instructions: state.DrinkInfo.instructions,
        ingredient1: state.DrinkInfo.ingredient1,
        ingredient2: state.DrinkInfo.ingredient2,
        ingredient3: state.DrinkInfo.ingredient3,
        picture: state.DrinkInfo.picture,
        measure1: state.DrinkInfo.measure1,
        measure2: state.DrinkInfo.measure2,
        measure3: state.DrinkInfo.measure3,
        isFave: state.DrinkInfo.isFave
    }
}

export default connect(mapStateToProps) (Random)
