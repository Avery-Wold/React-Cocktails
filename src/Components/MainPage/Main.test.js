import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, configure } from "enzyme";
import Header from "../Header";
import {CocktailSearch} from "../CocktailSearch";
import Random from "../Random";
import SpiritSearch from "../SpiritSearch";
import { Drink } from "../Drink";
import { Main } from "./Main";
import {getAveryFave} from "./Main"

configure({adapter: new Adapter()});

describe("Main", () => {
    it("header should render correctly", () => {
        const shallowApp = shallow(<Header />);
        expect(shallowApp).toMatchSnapshot();
    });

    it("cocktailsearch should render correctly", () => {
        const shallowApp = shallow(<CocktailSearch />);
        expect(shallowApp).toMatchSnapshot();
    });

    it("random should render correctly", () => {
        const shallowApp = shallow(<Random />);
        expect(shallowApp).toMatchSnapshot();
    });

    it("spiritsearch should render correctly", () => {
        const shallowApp = shallow(<SpiritSearch />);
        expect(shallowApp).toMatchSnapshot();
    });

    it("should find drink when averys fave clicked", () => {
        const mountedApp = mount(<Main />);
        mountedApp
          .find('btn-fave')
          .simulate('click')
        expect(mountedApp.find(Drink).length).toBe(1);
    });

    // it('should call get avery fave function when button is clicked', () => {
    //     const tree = shallow(
    //       <button name='btn-fave' handleClick={getAveryFave} />
    //     );
    //     tree.simulate('click');
    //     expect(getAveryFave).toHaveBeenCalled();
    //   });
});