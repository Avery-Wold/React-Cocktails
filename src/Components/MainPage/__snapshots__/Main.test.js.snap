// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main cocktailsearch should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(CocktailSearch) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Main header should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Example />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Navbar
        expand={false}
        tag="nav"
      >
        <NavbarBrand
          className="title"
          drinks=""
          href="/"
          tag="a"
        >
          Making Cocktails With Avery
        </NavbarBrand>
        <div
          className="pull-right"
        >
          <NavbarToggler
            className="navbar-collapse"
            onClick={[Function]}
            tag="button"
            type="button"
          />
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Nav
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                  tag="a"
                >
                  GitHub
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </div>
      </Navbar>,
      "className": "navbar navbar-expand-md bg-dark navbar-dark",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavbarBrand
            className="title"
            drinks=""
            href="/"
            tag="a"
          >
            Making Cocktails With Avery
          </NavbarBrand>,
          <div
            className="pull-right"
          >
            <NavbarToggler
              className="navbar-collapse"
              onClick={[Function]}
              tag="button"
              type="button"
            />
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Nav
                navbar={true}
                tag="ul"
                vertical={false}
              >
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                    tag="a"
                  >
                    GitHub
                  </NavLink>
                </NavItem>
              </Nav>
            </Collapse>
          </div>,
        ],
        "expand": false,
        "tag": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Making Cocktails With Avery",
            "className": "title",
            "drinks": "",
            "href": "/",
            "tag": "a",
          },
          "ref": null,
          "rendered": "Making Cocktails With Avery",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavbarToggler
                className="navbar-collapse"
                onClick={[Function]}
                tag="button"
                type="button"
              />,
              <Collapse
                appear={false}
                enter={true}
                exit={true}
                in={false}
                isOpen={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                tag="div"
                timeout={350}
                unmountOnExit={false}
              >
                <Nav
                  navbar={true}
                  tag="ul"
                  vertical={false}
                >
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                      tag="a"
                    >
                      GitHub
                    </NavLink>
                  </NavItem>
                </Nav>
              </Collapse>,
            ],
            "className": "pull-right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "navbar-collapse",
                "onClick": [Function],
                "tag": "button",
                "type": "button",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "appear": false,
                "children": <Nav
                  navbar={true}
                  tag="ul"
                  vertical={false}
                >
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                      tag="a"
                    >
                      GitHub
                    </NavLink>
                  </NavItem>
                </Nav>,
                "enter": true,
                "exit": true,
                "in": false,
                "isOpen": false,
                "mountOnEnter": false,
                "onEnter": [Function],
                "onEntered": [Function],
                "onEntering": [Function],
                "onExit": [Function],
                "onExited": [Function],
                "onExiting": [Function],
                "tag": "div",
                "timeout": 350,
                "unmountOnExit": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <NavItem
                    tag="li"
                  >
                    <NavLink
                      href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                      tag="a"
                    >
                      GitHub
                    </NavLink>
                  </NavItem>,
                  "navbar": true,
                  "tag": "ul",
                  "vertical": false,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavLink
                      href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                      tag="a"
                    >
                      GitHub
                    </NavLink>,
                    "tag": "li",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "GitHub",
                      "href": "https://github.com/Avery-Wold/React-Masters-Spring-2019",
                      "tag": "a",
                    },
                    "ref": null,
                    "rendered": "GitHub",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Navbar
          expand={false}
          tag="nav"
        >
          <NavbarBrand
            className="title"
            drinks=""
            href="/"
            tag="a"
          >
            Making Cocktails With Avery
          </NavbarBrand>
          <div
            className="pull-right"
          >
            <NavbarToggler
              className="navbar-collapse"
              onClick={[Function]}
              tag="button"
              type="button"
            />
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Nav
                navbar={true}
                tag="ul"
                vertical={false}
              >
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                    tag="a"
                  >
                    GitHub
                  </NavLink>
                </NavItem>
              </Nav>
            </Collapse>
          </div>
        </Navbar>,
        "className": "navbar navbar-expand-md bg-dark navbar-dark",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <NavbarBrand
              className="title"
              drinks=""
              href="/"
              tag="a"
            >
              Making Cocktails With Avery
            </NavbarBrand>,
            <div
              className="pull-right"
            >
              <NavbarToggler
                className="navbar-collapse"
                onClick={[Function]}
                tag="button"
                type="button"
              />
              <Collapse
                appear={false}
                enter={true}
                exit={true}
                in={false}
                isOpen={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                tag="div"
                timeout={350}
                unmountOnExit={false}
              >
                <Nav
                  navbar={true}
                  tag="ul"
                  vertical={false}
                >
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                      tag="a"
                    >
                      GitHub
                    </NavLink>
                  </NavItem>
                </Nav>
              </Collapse>
            </div>,
          ],
          "expand": false,
          "tag": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Making Cocktails With Avery",
              "className": "title",
              "drinks": "",
              "href": "/",
              "tag": "a",
            },
            "ref": null,
            "rendered": "Making Cocktails With Avery",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavbarToggler
                  className="navbar-collapse"
                  onClick={[Function]}
                  tag="button"
                  type="button"
                />,
                <Collapse
                  appear={false}
                  enter={true}
                  exit={true}
                  in={false}
                  isOpen={false}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  tag="div"
                  timeout={350}
                  unmountOnExit={false}
                >
                  <Nav
                    navbar={true}
                    tag="ul"
                    vertical={false}
                  >
                    <NavItem
                      tag="li"
                    >
                      <NavLink
                        href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                        tag="a"
                      >
                        GitHub
                      </NavLink>
                    </NavItem>
                  </Nav>
                </Collapse>,
              ],
              "className": "pull-right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "navbar-collapse",
                  "onClick": [Function],
                  "tag": "button",
                  "type": "button",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "appear": false,
                  "children": <Nav
                    navbar={true}
                    tag="ul"
                    vertical={false}
                  >
                    <NavItem
                      tag="li"
                    >
                      <NavLink
                        href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                        tag="a"
                      >
                        GitHub
                      </NavLink>
                    </NavItem>
                  </Nav>,
                  "enter": true,
                  "exit": true,
                  "in": false,
                  "isOpen": false,
                  "mountOnEnter": false,
                  "onEnter": [Function],
                  "onEntered": [Function],
                  "onEntering": [Function],
                  "onExit": [Function],
                  "onExited": [Function],
                  "onExiting": [Function],
                  "tag": "div",
                  "timeout": 350,
                  "unmountOnExit": false,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavItem
                      tag="li"
                    >
                      <NavLink
                        href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                        tag="a"
                      >
                        GitHub
                      </NavLink>
                    </NavItem>,
                    "navbar": true,
                    "tag": "ul",
                    "vertical": false,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <NavLink
                        href="https://github.com/Avery-Wold/React-Masters-Spring-2019"
                        tag="a"
                      >
                        GitHub
                      </NavLink>,
                      "tag": "li",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "GitHub",
                        "href": "https://github.com/Avery-Wold/React-Masters-Spring-2019",
                        "tag": "a",
                      },
                      "ref": null,
                      "rendered": "GitHub",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Main main should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(CocktailSearch) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Main random should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Random) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Main spiritsearch should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(SpiritSearch) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
